server/src/test/java/org/spine3/server/entity/storagefield/StorageFieldsExtractorShould.java

```diff
@@ -0,0 +1,203 @@
+
+        verify(entity).getId();
+        verify(entity).getFloatNull();
+        verify(entity).getIntegerFieldValue();
+        verify(entity).getSomeMessage();
+
+        verify(entity, never()).getSomeNonPublicMethod();
+        verify(entity, never()).getSomeUnsupportedType();
+        verify(entity, never()).getSomeVoid();
+
+        checkId(entity, fields);
+
+        // Message fields map contains the only custom field
+        final Map<String, Any> anyFields = fields.getAnyFieldMap();
+        assertSize(1, anyFields);
+        final Any messageField = anyFields.get("somemessage");
+        assertNotNull(messageField);
+        assertEquals(entity.getSomeMessage(), AnyPacker.unpack(messageField));
+
+        // Float field map contains no entries (as field in null)
+        final Map<String, Float> floatFields = fields.getFloatFieldMap();
+        assertEmpty(floatFields);
+
+        // Integer field map containt the only entry
+        final Map<String, Integer> intFields = fields.getIntegerFieldMap();
+        assertSize(1, intFields);
+        final Integer field = intFields.get("IntegerFieldValue".toLowerCase());
+        assertNotNull(field);
+        assertEquals(entity.getIntegerFieldValue(), field.intValue());
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void throw_if_non_null_method_returns_null() {
+        StorageFieldsExtractor.extract(new EntityWithInvalidGetters(STRING_ID));
+    }
+
+    @Test
+    public void hadnle_non_public_entity_class() {
+        final Optional<StorageFields> fields =
+                StorageFieldsExtractor.extract(new PrivateEntity(STRING_ID));
+        assertFalse(fields.isPresent());
+    }
+
+    private static void checkId(Entity entity, StorageFields fields) {
+        @SuppressWarnings("OverlyStrongTypeCast") // ...OrBuilder
+        final String id = ((StringValue) AnyPacker.unpack(fields.getEntityId())).getValue();
+        final String entityId = (String) entity.getId();
+        assertEquals("StorageFields contain not the same entity ID as passed",
+                     entityId,
+                     id);
+    }
+
+    public static class EntityWithNoStorageFields extends AbstractEntity<String, Any> {
+        protected EntityWithNoStorageFields(String id) {
+            super(id);
+        }
+    }
+
+    public static class EntityWithManyGetters extends AbstractEntity<String, Any> {
+
+        private final Message someMessage = Sample.messageOfType(Project.class);
+
+        protected EntityWithManyGetters(String id) {
+            super(id);
+        }
+
+        public int getIntegerFieldValue() {
+            return 0;
+        }
+
+        @Nullable
+        public Float getFloatNull() {
+            return null;
+        }
+
+        public Message getSomeMessage() {
+            return someMessage;
+        }
+
+        int getSomeNonPublicMethod() {
+            throw new AssertionError("getSomeNonPublicMethod invoked");
```

Comment:
This is a test code.
I've chosen the `AssertionError` to make it clear we __do not__ handle is anywhere in the code.